### ISSUE TITLE
legacy render is not working on nixos(flakes)(black screen)

### ISSUE THREAD

**Issue by winder2412** [hyprwm/Hyprland#5254]
### Hyprland Version

i can't run "hyprctl systeminfo" because i can't run hyprland 
Version:
it says 37.1 when it's rebuilding the system but in the /tmp/hypr directory it says v0.36.0

### Bug or Regression?

Bug

### Description

i have a laptop that is pretty old 
i had hyprland on it on arch (i compiled it manually using legacy render)
i installed nixos (unstable) on it and tried to install hyprland with legacy render on the laptop(using flakes) and after it builds and i run it, it would display a black screen(like when i install hyprland without the legacy render set to ture)

### How to reproduce

1- on a fresh install of nixos(on an old laptop) enable flakes

2- add hyprland url in the inputs of your flakes 

3- add this to your configuration.nix :
`
programs.hyprland = {
    enable = true;
    package = inputs.hyprland.packages.${pkgs.system}.hyprland.override { legacyRenderer = true; };
    xwayland.enable = true;
  };
`

4- run Hyprland and it would show a black screen

### Crash reports, logs, images, videos

there is no crash report in the log

`[LOG] Instance Signature: v0.36.0_1711328463
[LOG] Hyprland PID: 16847
[LOG] ===== SYSTEM INFO: =====
[LOG] System name: Linux
[LOG] Node name: nixos
[LOG] Release: 6.1.78
[LOG] Version: #1-NixOS SMP PREEMPT_DYNAMIC Fri Feb 16 18:06:32 UTC 2024

[LOG] GPU information:
00:02.0 VGA compatible controller [0300]: Intel Corporation Mobile 4 Series Chipset Integrated Graphics Controller [8086:2a42] (rev 07) (prog-if 00 [VGA controller])

[LOG] os-release:
ANSI_COLOR="1;34"
BUG_REPORT_URL="https://github.com/NixOS/nixpkgs/issues"
BUILD_ID="24.05.20240319.b06025f"
DOCUMENTATION_URL="https://nixos.org/learn.html"
HOME_URL="https://nixos.org/"
ID=nixos
IMAGE_ID=""
IMAGE_VERSION=""
LOGO="nix-snowflake"
NAME=NixOS
PRETTY_NAME="NixOS 24.05 (Uakari)"
SUPPORT_URL="https://nixos.org/community.html"
VERSION="24.05 (Uakari)"
VERSION_CODENAME=uakari
VERSION_ID="24.05"

[LOG] ========================

[INFO] If you are crashing, or encounter any bugs, please consult https://wiki.hyprland.org/Crashes-and-Bugs/

[LOG]
Current splash: "@vaxry how do i learn c++" - flicko

[LOG] Creating the HookSystem!
[LOG] Creating the KeybindManager!
[LOG] [hookSystem] New hook event registered: configReloaded
[LOG] Creating the AnimationManager!
[LOG] Created a bezier curve, baked 255 points, mem usage: 4.08kB, time to bake: 96.11µs. Estimated average calc time: 0.34µs.
[LOG] Creating the ConfigManager!
[LOG] NOTE: further logs to stdout / logfile are disabled by default. Use debug:disable_logs and debug:enable_stdout_logs to override this.
[LOG] Created a bezier curve, baked 255 points, mem usage: 4.08kB, time to bake: 79.63µs. Estimated average calc time: 0.34µs.
[LOG] Using config: /home/winder/.config/hypr/hyprland.conf`

**Comment by fufexan** [comment]
0.36.0 looks like the version currently in Nixpkgs unstable (not sure if the 0.37.1 update made it through yet).
Can you try overriding the Nixpkgs package instead of the flake one?

**Comment by winder2412** [comment]
i tried that but it fails to build it 
this is the error log:

error: builder for '/nix/store/4ha3vscw0bwlja3a180x498kkn1zs66w-hyprland-0.36.0.drv' failed with exit code 1;
       last 10 log lines:
       >   356 |     if (ERR == GL_CONTEXT_LOST) /* We don't have infra to recover from this */
       >       |                ^~~~~~~~~~~~~~~
       >       |                EGL_CONTEXT_LOST
       > ../src/render/OpenGL.cpp: In member function 'void CHyprOpenGLImpl::createBGTextureForMonitor(CMonitor*)':
       > ../src/render/OpenGL.cpp:2051:66: error: 'GL_RGB32F' was not declared in this scope; did you mean 'GL_RGB565'?
       >  2051 |     const GLint glIFormat = CAIROFORMAT == CAIRO_FORMAT_RGB96F ? GL_RGB32F : GL_RGBA;
       >       |                                                                  ^~~~~~~~~
       >       |                                                                  GL_RGB565
       > [116/132] Compiling C++ object src/Hyprland.p/render_Framebuffer.cpp.o
       > ninja: build stopped: subcommand failed.
       For full logs, run 'nix log /nix/store/4ha3vscw0bwlja3a180x498kkn1zs66w-hyprland-0.36.0.drv'.

**Comment by fufexan** [comment]
Okay, I believe that failure was fixed in 0.37.1 since it builds fine. Could you try overriding the mesa package as well?

```nix
  programs.hyprland = {
    enable = true;
    package = inputs.hyprland.packages.${pkgs.system}.hyprland.override {
      legacyRenderer = true;
      mesa = pkgs.mesa;
    };
    xwayland.enable = true;
  };
```

**Comment by winder2412** [comment]
So i figured that the problem was that i did not add the `package = inputs.hyprland.packages.${pkgs.system}.hyprland.override` in the home-manager config so i guess that the home-manager was reseting it to default Nixpkgs version which was version 0.36 and that was the reason that in the /tmp/hyprland i only had v0.36

so it got fixed when i put the overlay both on the main configuration and in the home-manager
